pkg/network/usm/sharedlibraries/ebpf.go

```diff
@@ -379,15 +420,18 @@ func (e *EbpfProgram) stopImpl() {
 		ebpftelemetry.UnregisterTelemetry(e.Manager.Manager)
 	}
 
-	for _, data := range e.libsets {
-		data.perfHandler.Stop()
-		close(data.done)
+	for _, handler := range e.libsets {
+		if handler.enabled {
+			handler.stop()
```

Comment:
shouldn't we delete the libsets or mark them as `not enabled` after stopping them?

Reply:
Changed the `stop` function of the handler to set itself as disabled